zos_concepts/templates/copy_template/ansible.cfg

```diff
@@ -0,0 +1,19 @@
+################################################################################
+# Copyright (c) IBM Corporation 2023
+################################################################################
+
+################################################################################
+# Minimum required configuration for this playbook.
+# For additional `ansible.cfg` configuration options, review:
+#   https://docs.ansible.com/ansible/latest/reference_appendices/config.html
+#   https://github.com/ansible/ansible/blob/devel/examples/ansible.cfg
+################################################################################
+
+[defaults]
+forks = 25
+roles_path = ../../../roles:.
+stdout_callback = unixy
```

Comment:
Curious about stdout_callback = unixy. Could you explain?

Reply:
Sure. [`stdout_callback`](https://docs.ansible.com/ansible/latest/plugins/callback.html#callback-plugins) tells Ansible how it should format the output from a playbook, there's a whole bunch of callbacks available:
```
>ansible-doc -t callback -l

amazon.aws.aws_resource_actions       summarizes all "resource:actions" completed                                                                                                   
ansible.builtin.default               default Ansible screen output                                                                                                                 
ansible.builtin.junit                 write playbook output to a JUnit file                                                                                                         
ansible.builtin.minimal               minimal Ansible screen output                                                                                                                 
ansible.builtin.oneline               oneline Ansible screen output                                                                                                                 
ansible.builtin.tree                  Save host events to files                                                                                                                     
ansible.posix.cgroup_perf_recap       Profiles system activity of tasks and full execution using cgroups                                                                            
ansible.posix.debug                   formatted stdout/stderr display                                                                                                               
ansible.posix.json                    Ansible screen output as JSON                                                                                                                 
ansible.posix.jsonl                   Ansible screen output as JSONL (lines in json format)                                                                                         
ansible.posix.profile_roles           adds timing information to roles                                                                                                              
ansible.posix.profile_tasks           adds time information to tasks                                                                                                                
ansible.posix.skippy                  Ansible screen output that ignores skipped status                                                                                             
ansible.posix.timer                   Adds time to play stats                                                                                                                       
community.general.cgroup_memory_recap Profiles maximum memory usage of tasks and full execution using cgroups                                                                       
community.general.context_demo        demo callback that adds play/task context                                                                                                     
community.general.counter_enabled     adds counters to the output items (tasks and hosts/task)                                                                                      
community.general.dense               minimal stdout output                                                                                                                         
community.general.diy                 Customize the output                                                                                                                          
community.general.elastic             Create distributed traces for each Ansible task in Elastic APM                                                                                
community.general.hipchat             post task events to hipchat                                                                                                                   
community.general.jabber              post task events to a jabber server                                                                                                           
community.general.log_plays           write playbook output to log file                                                                                                             
community.general.loganalytics        Posts task results to Azure Log Analytics                                                                                                     
community.general.logdna              Sends playbook logs to LogDNA                                                                                                                 
community.general.logentries          Sends events to Logentries                                                                                                                    
community.general.logstash            Sends events to Logstash                                                                                                                      
community.general.mail                Sends failure events via email                                                                                                                
community.general.nrdp                Post task results to a Nagios server through nrdp                                                                                             
community.general.null                Don't display stuff to screen                                                                                                                 
community.general.opentelemetry       Create distributed traces with OpenTelemetry                                                                                                  
community.general.say                 notify using software speech synthesizer                                                                                                      
community.general.selective           only print certain tasks                                                                                                                      
community.general.slack               Sends play events to a Slack channel                                                                                                          
community.general.splunk              Sends task result events to Splunk HTTP Event Collector                                                                                       
community.general.sumologic           Sends task result events to Sumologic                                                                                                         
community.general.syslog_json         sends JSON events to syslog                                                                                                                   
community.general.unixy               condensed Ansible output                                                                                                                      
community.general.yaml                YAML-ized Ansible screen output                                                                                                               
community.grafana.grafana_annotations send ansible events as annotations on charts to grafana over http api                                                                         
ovirt.ovirt.stdout                    Output the log of ansible                                                                                                                     
theforeman.foreman.foreman            Sends events to Foreman
```

During development for the template support, people wondered how whitespace was treated by Jinja when it substitutes a block in a template, and whether that could introduce spaces or newlines that can break things, like in JCL or in data sets. The last tasks showcasing `zos_copy` highlight how `trim_blocks` and `keep_trailing_newline` can affect how a template is rendered and copied to a remote system. So these tasks tell Jinja to render a template with a bunch of unneeded newlines and then copy that to a data set, creating empty records.

In the default output format, this looks like this:
```json
{
    "msg": {
        "changed": true,
        "cmd": [
            "cat",
            "//'OMVSADM.P0000460.T0238546.C0000000'"
        ],
        "delta": "0:00:00.219972",
        "end": "2023-07-19 22:00:40.757593",
        "failed": false,
        "msg": "",
        "rc": 0,
        "start": "2023-07-19 22:00:40.537621",
        "stderr": "",
        "stderr_lines": [],
        "stdout": "00000001 First record\n\n00000002 This a record\n\n\n00000003 Value 6 passed the test\n\n\n00000004 Record from a loop\n\n00000005 Record from a loop\n\n00000006 Record from a loop",
        "stdout_lines": [
            "00000001 First record",
            "",
            "00000002 This a record",
            "",
            "",
            "00000003 Value 6 passed the test",
            "",
            "",
            "00000004 Record from a loop",
            "",
            "00000005 Record from a loop",
            "",
            "00000006 Record from a loop"
        ]
    }
}
```

Which didn't convince me it would be clear enough to a user what is happening. Using `unixy`, it looks like this:

```
See the contents of OMVSADM.P0000012.T0468865.C0000000. Note the blank lines....
  zos_host done | stdout: 00000001 First record

00000002 This a record


00000003 Value 6 passed the test


00000004 Record from a loop

00000005 Record from a loop

00000006 Record from a loop
```

Which I think looks a lot cleaner. If you have any more questions, shoot them my way

Reply:
@rexemin @fernandofloresg 

My initial run of the playbook failed with:
```
(ansible-2.11) ketan@laptop copy_template % ansible-playbook -i inventories zos_copy_template.yml 
ERROR! Invalid callback for stdout specified: unixy
```

To resolve this, I had to ensure that `unixy` was installed on my system (controller). `unixy` is part of the `community.general` package:
```
(ansible-2.11) ketan@laptop copy_template % ansible-galaxy collection install community.general
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Downloading https://galaxy.ansible.com/download/community-general-7.2.1.tar.gz to /Users/ketan/.ansible/tmp/ansible-local-40062zv83i3zn/tmp49nsnos6/community-general-7.2.1-ejqjr4rn
Installing 'community.general:7.2.1' to '/Users/ketan/.ansible/collections/ansible_collections/community/general'
community.general:7.2.1 was installed successfully
```


This would be worth mentioning in the playbook docs somewhere, maybe in the READMEs for each playbook?